src/System.Linq/tests/AggregateTests.cs

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+using System;
+using System.Linq;
+
+namespace Test
+{
+    public class AggregateTests
+    {
+        //
+        // Aggregate
+        //
+        [Fact]
+        public static void RunAggregationTests()
+        {
+            RunAggregationTest1_Sum1(1024);
+            RunAggregationTest1_Sum2(1024);
+            RunAggregationTest1_Sum3(1024);
+            RunAggregationTest2(16);
+        }
+
+        private static void RunAggregationTest1_Sum1(int count)
+        {
+            int expectSum = 0;
+            int[] ints = new int[count];
+            for (int i = 0; i < ints.Length; i++)
+            {
+                ints[i] = i;
+                expectSum += i;
+            }
+
+            int realSum = ints.Aggregate<int>(
+                delegate (int x, int y) { return x + y; });
+            if (realSum != expectSum)
+                Assert.True(false, string.Format("RunAggregationTest1_Sum1: FAILED. (count: {2}):  FAIL  > Expect: {0}, real: {1}", expectSum, realSum, count));
```

Comment:
FWIW, I agree with @Maxwe11, here.  These are effectively new tests, as they're copy-and-pasted with some changes, and fixing the PLINQ ones (which I believe @Clockwork-Muse is working on separately) won't also then fix these.  As such, we should treat these as new tests being written, and write them the way we think they should be rewritten the first time.